### PR TITLE
feat: add udt type handler

### PR DESCRIPTION
**Pull request by shaojunda**

### CODE REVIEW

File: lib/ckb/config.rb [hunk 1]
```diff
@@ -8,16 +8,23 @@ class Config
     SECP256K1_BLAKE160_SIGHASH_ALL_TYPE_HASH = "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8"
     # https://github.com/nervosnetwork/ckb/blob/develop/resource/specs/mainnet.toml#L127
     SECP256K1_BLAKE160_MULTISIG_ALL_TYPE_HASH = "0x5c5069eb0857efc65e1bca0c07df34c31663b3622fd3876c876320fc9634e2a8"
+    # This is my locally deployed sudt type script's code hash. This code hash will be replaced after the real sudt type script deployed on mainnet in the future
+    SUDT_CODE_HASH = "0x48dbf59b4c7ee1547238021b4869bceedf4eea6b43772e5d66ef8865b6ae7212"
+    SUDT_TX_HASH = "0x0f18ac6058f7e1cef8a94b4709be58077aef1ad586403c4337226af3fb12ba29"
```
Comment (duanyytop) [inline lib/ckb/config.rb:13]: Should we give two tx _hashes which include mainnet and aggron ?

File: lib/ckb/lock_handlers/anyone_can_pay_handler.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+module CKB
+  module LockHandlers
+    class AnyoneCanPayHandler < SingleSignHandler
+      attr_reader :anyone_can_pay_cell_dep
+
+      def initialize(anyone_can_pay_tx_hash = "0xeba5986c50e7e8215c2acd6501150cab1346119f42f4d7870bfadd5bfdf7fe57", index = 0)
```
Comment (duanyytop) [inline lib/ckb/lock_handlers/anyone_can_pay_handler.rb:6]: Maybe set `anyone_can_pay_tx_hash ` will be more better ?

File: lib/ckb/wallets/anyone_can_pay_wallet.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+          capacity = is_owner && output_info[:type] && output_info[:type].compute_hash == sudt_type_script.compute_hash ? [output_info[:capacity], CKB::Utils.byte_to_shannon(142)].max : output_info[:capacity]
+          outputs << CKB::Types::Output.new(capacity: capacity, lock: lock, type: output_info[:type])
+          outputs_data << (output_info[:data] || "0x")
+        end
+
+        if outputs.all? { |output| output.capacity > 0 } || (outputs.map { |output| output.capacity }.sum == 0)
+          if to_infos.any? { |_, output_info| output_info[:type] && output_info[:type].compute_hash == sudt_type_script.compute_hash }
+            outputs << CKB::Types::Output.new(capacity: 0, lock: input_scripts[-1], type: nil)
+            outputs_data << "0x"
+            outputs << CKB::Types::Output.new(capacity: CKB::Utils.byte_to_shannon(142), lock: input_scripts[-1], type: sudt_type_script)
```
Comment (duanyytop) [inline lib/ckb/wallets/anyone_can_pay_wallet.rb:64]: 142 to a const ?

Comment (shaojunda) [reply lib/ckb/wallets/anyone_can_pay_wallet.rb]: changed. https://github.com/nervosnetwork/ckb-sdk-ruby/pull/244/commits/90aa364444e449b2c9b87c6448085856a4d92cad

File: lib/ckb/lock_handlers/anyone_can_pay_handler.rb (hunk 2 above)
Comment (shaojunda) [reply lib/ckb/lock_handlers/anyone_can_pay_handler.rb]: I decided to change to manually set lock and type that are not built into the system

**Pull request merged**